### ISSUE TITLE
In new pages: Ensure all illustration-related images have helpful "alt" text

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#259]
_Part of epic: #257_

As a website content creator adding an image to a page, I will be forced to provide "alt" text content when adding the image.

**Comment by paregorios** [comment]
@cguardia @skleinfeldt I assume we're going to need to punt this ticket into the next iteration, yes?

**Comment by cguardia** [comment]
I have two hours for this week. I was going to use them to try and finish this one. Is that OK?

**Comment by paregorios** [comment]
sounds good!

**Comment by cguardia** [comment]
I spent some time looking at this. I also started working on a branch that overrides the editor image chooser to force the alt text field for images. I'm close, but I have a few points that I thought I should bring up.

First, adding or uploading an image into the site, relies on the image properties. The title attribute is taken from the image object created inside plone, and is not editable after creating the image. Will we want to make the title attribute editable after image creation? Note that every other page that uses this same image will have the same title attribute.

The image description is used for the caption text, which in turn goes into the alt attribute. The description can be updated, but then every other page that uses this image will get the updated description too. It's not possible to have different alt text for the same image in different pages.

If what I described above is good enough for us, then all I need to do is make the caption field required when uploading or inserting an image. Although I didn't finish the ticket this week, I can finish it on Monday if this is the case.

If we require a different behavior, this would probably require a bit more work.

**Comment by cguardia** [comment]
Waiting for feedback on this one.

**Comment by paregorios** [comment]
I will discuss with @k2soule and provide feedback asap.

**Comment by jessesnyder** [comment]
@paregorios @k2soule My friend Mark, who's quite passionate about accessibility, pointed me to a decision tree for deciding alt text produced by the w3c. Here's his blog post on the issue, which includes a link to the w3c article: https://mrwweb.com/alt-text-decision-tree/

**Comment by paregorios** [comment]
@jessesnyder thanks!

**Comment by paregorios** [comment]
@cguardia I think we have figured out what we want:

## recommendations

1. never create "title" attribute on an html:img or other tag wrapping an image
2. always wrap an embedded image in an html:a link (currently only occurs when caption=true)
3. create new functionality that requires a user to enter text value for the img@alt attribute each time an image is embedded somewhere
    - cannot be empty
    - maximum length: 150 char
    - if same image is embedded multiple times/places, it can have different img@alt value each place it is embedded (i.e., this is **not** a new property on the image object)
4. create new functionality that requires a user to enter text value for a caption when an image is embedded with caption=true
    - cannot be empty
    - no maximum length?
    - if same image is embedded multiple times/places, it can have different caption value each place it is embedded (i.e., this is **not** a new property on the image object)


## example when caption=false

```html
<span class="image-inline" style="display: block; width:128px;">
    <a rel="lightbox"
       href="path/to/image/object">
        <img src="path/to/image/object/@@images/etcblahblah">
             alt="Photo of a banjo in a person's hands. Printing on the head reads: this machine surrounds hate and forces it to surrender."
             width="128"
             height="128">
    </a>
</span>
```

## example when caption=true

```html
<span class="image-inline" style="display: block; width:128px;">
    <span>
        <a rel="lightbox"
           href="path/to/image/object">
            <img src="path/to/image/object/@@images/etcblahblah">
                 alt="Photo of a banjo in a person's hands. Printing on the head reads: this machine surrounds hate and forces it to surrender."
                 width="128"
                 height="128">
        </a>
    </span>
    <span class="image-caption" style="display: block; width:128px;">
        A close-up of Pete Seeger's banjo, showing its famous slogan. Photo date and photographer unknown.
    </span>
</span>

**Comment by cguardia** [comment]
Ok, so we *never* will use the title attribute for lead image img tags? This would have implications for previous work related to this (nothing huge, though).

**Comment by cguardia** [comment]
Please reply to my last comment as soon as possible, since it will affect the way I code this.

**Comment by paregorios** [comment]
@cguardia correct: **never** use title attribute on img tags, whether they are put inline through the rich text editor or via lead image.

**Comment by cguardia** [comment]
This is on staging now. I removed the caption checkbox, and just use the caption if the text is there, and skip it if the text is empty.

**Comment by paregorios** [comment]
- @cguardia "alt" and "title" behaviors are now working as expected
 - captioning is behaving well

There are some other concerns that I laid out above that are not yet working as desired, but I want to go ahead and get the current functionality deployed to production. I will create new, separate tickets for the additional behaviors that have not yet been implemented.

Please deploy the code that is currently running on staging for this feature to production.
